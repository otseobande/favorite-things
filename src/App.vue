<template>
  <div id="app">
    <div class="pl-32">
      <Header />
      <div class="flex pt-10">
        <Category />
        <Category />
        <div class="h-12 max-h-full w-64 p-3 m-0 mr-4 bg-gray-500 hover:bg-gray-600 rounded text-white font-bold cursor-pointer outline-none">
          <button>
            <img width="20" class="inline mr-3" src="./assets/add-icon-white.svg" />
            <span>Add new category</span>
          </button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Header from './components/Header.vue';
import Category from './components/Category.vue'

export default {
  name: 'app',
  components: {
    Header,
    Category
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;

}

html, body {
  height: 100%;
  background: linear-gradient(90deg, #4b6cb7 0%, #182848 100%);
}
</style>
