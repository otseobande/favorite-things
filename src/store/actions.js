import * as actionTypes from './actions.type';
import * as mutationTypes from './mutations.type';

import categoryService from '../services/categoryService';
import favoriteThingService from '../services/favoriteThingService';

export default {
  async [actionTypes.FETCH_CATEGORIES]({ commit }) {
    const response = await categoryService.get();

    commit(mutationTypes.SET_CATEGORIES, response.data.results)
  },

  async [actionTypes.ADD_FAVORITE_THING]({ commit }, newFavoriteThing) {
    const response = await favoriteThingService.create(newFavoriteThing);

    commit(mutationTypes.ADD_FAVORITE_THING, response.data)
  },

  async [actionTypes.UPDATE_FAVORITE_THING]({ commit }, {
    favoriteThingId,
    updatedFavoriteThingData
  }) {
    const response = await favoriteThingService.update({
      favoriteThingId,
      updatedFavoriteThingData
    })

    commit(mutationTypes.UPDATE_FAVORITE_THING, response.data);
  },

  async [actionTypes.DELETE_FAVORITE_THING]({ commit }, { favoriteThingId }) {
    await favoriteThingService.delete({
      favoriteThingId
    });

    commit(mutationTypes.DELETE_FAVORITE_THING, favoriteThingId)
  },

  async [actionTypes.ADD_DUMMY_CATEGORY]({ commit }) {
    commit(mutationTypes.ADD_CATEGORY, {
      id: 'template',
      name: '',
      favorite_things: []
    })
  },

  async [actionTypes.DELETE_DUMMY_CATEGORY]({ commit }) {
    commit(mutationTypes.DELETE_CATEGORY, 'template')
  },

  async [actionTypes.ADD_CATEGORY]({ commit }, newCategory) {
    const response = await categoryService.create(newCategory)

    commit(mutationTypes.ADD_CATEGORY, response.data)
  },

  async [actionTypes.DELETE_CATEGORY]({ commit }, { categoryId }) {
    await categoryService.delete(categoryId);

    commit(mutationTypes.DELETE_CATEGORY, categoryId);
  }
}
