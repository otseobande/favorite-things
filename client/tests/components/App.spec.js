import { shallowMount, createLocalVue } from '@vue/test-utils';
import App from '../../src/App.vue';
import VueTailwind from 'vue-tailwind';
import Vuex from 'vuex';
import actions from '../../src/store/actions';
import * as actionTypes from '../../src/store/actions.type';

jest.mock('../../src/store/actions');

const localVue = createLocalVue();

localVue.use(VueTailwind);
localVue.use(Vuex);

let wrapper;

describe('App component', () => {
  beforeEach(() => {
    wrapper = shallowMount(App, {
      localVue,
      store: new Vuex.Store({
        actions
      }),
    });
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('addDummyCategory', () => {
    it('should call ADD_DUMMY_CATEGORY method', () => {
      wrapper.vm[actionTypes.ADD_DUMMY_CATEGORY] = jest.fn();

      wrapper.vm.addDummyCategory()

      expect(wrapper.vm[actionTypes.ADD_DUMMY_CATEGORY]).toBeCalled()
    })
  })

  it('should fetch categories on mount', () => {
    expect(actions[actionTypes.FETCH_CATEGORIES]).toBeCalled()
  })
})
