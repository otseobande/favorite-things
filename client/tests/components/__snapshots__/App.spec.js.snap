// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div id="app">
  <div class="pl-24">
    <header-stub></header-stub>
    <div class="flex flex-row items-start pt-10 mb-10 pr-10"> <button class="
        h-12
        max-h-full
        w-64
        p-3
        m-0
        mr-10
        bg-gray-500
        hover:bg-gray-600
        rounded
        text-white
        font-bold
        cursor-pointer
        outline-none
        category"><img width="20" src="./assets/add-icon-white.svg" class="inline mr-3"> <span class="font-bold">Add new category</span></button></div>
  </div>
</div>
`;
